doc/src/hugo/content/guides/authors/design/patterns/command-event/optional-parameters.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "3 - Optional Parameters"
+description: "Optional Version of Entity"
+date: 2022-02-25T10:50:32-07:00
+draft: "false"
+weight: 10
+---
+
+# Optional Versioning
+
+## Q: Should the event for each command have a version of the entity where all fields are optional,
+such that only changed data is sent back to gateway?
+
+### Example:
+
+type InfoUpdated {
+    name: String?,
+    address: Address?,
+    members: MemberId*,
+    isPrivate: Boolean?,
+}
+
+type ContactsUpdated is {
+    primaryContacts: MemberId+,
+    billingContacts: MemberId*,
+    distributionContacts: MemberId*
+}
+
+event OrganizationEstablished {
+    id: OrganizationId
+    info: Info,
+    contacts: Contacts,
+}
+
+event OrganizationInfoModified {
+    id: OrganizationId
+    info: InfoUpdated,
+}
+
+event UserContactsModified {
+    id: OrganizationId
+    contacts: ContactsUpdated,
+}
+
+## Suggestions
+
+This both removes clutter and makes it easier to see exactly what is going on in each event. We are excluding the Contacts object 
+from the event since it there is no way it will change when an Info field is changed. We are also using InfoUpdated and 
+ContactsUpdated in place of the original objects so we can also reduce our load - if we are modifying Info.name or Info.isPrivate,
+we don't need to bring the more complex & hefty Address object along with it because the field is now optional.
+
+These principles apply especially to object like Info that describe the basic details of an entity - they usually have
+many more fields than shown in this example, a mixture of `collection`s, [RIDDL Compound Types]({{< relref "concepts/type.md#compounds" >}}), and [RIDDL Predefined Types]({{< relref "language/common/types.md#Predefined Types" >}}).
+
+Any client consuming this event would have to know that the data coming back is not the exact copy of the data but only the fields
+which have been updated. We believe this is an easy enough inference to make for most developers.
```

Comment:
Developers are not the only consumer of these events. Business analysts, subject matter experts, etc. will all wonder about this because it isn't obvious, except to coders.